### PR TITLE
Describe three-valued logic in docstrings

### PR DESCRIPTION
**Pull request by nalimilan**
Add information regarding 3VL in the presence of missing values to relevant docstrings, in addition to what already exists in the manual. Also improve docstrings for comparison operators and make them more consistent with one another. Introduce a new `# Implementation` section intended at developers of custom types, and document it in the guidelines.

### CODE REVIEW

File: base/reduce.jl [hunk 1]
```diff
@@ -702,12 +742,19 @@ end
...
+Determine whether an item is in the given collection, in the sense that it is
+[`==`](@ref) to one of the values generated by iterating over the collection.
+Returns a `Bool` value, except if `item` is [`missing`](@ref) or `collection`
+contains `missing` but not `item`, in which case `missing` is returned
+([three-valued logic](https://en.wikipedia.org/wiki/Three-valued_logic),
+matching the behavior of [`any`](@ref) and [`==`](@ref)).
+
+Some collections follow a slightly different definition. For example,
+[`Set`](@ref)s check whether the item [`isequal`](@ref) to one of the elements.
+[`Dict`](@ref)s look for `(key,value)` pairs, and the key is compared using
```
Comment (JeffBezanson) [inline base/reduce.jl]: `key=>value`

File: doc/src/manual/documentation.md [hunk 2]
```diff
@@ -182,6 +182,11 @@ As in the example above, we recommend following some simple conventions when wri
 
    Docstrings are edited using the same tools as code. Therefore, the same conventions should apply.
    It it advised to add line breaks after 92 characters.
+6. Provide information allowing custom types to implement the function in an
+   `# Implementation` section. These implementation details intended at developers
```
Comment (JeffBezanson) [inline doc/src/manual/documentation.md]: "intended for developers"

**Review by JeffBezanson — APPROVED** [review]

**Comment by JeffBezanson** [comment]
Excellent, thank you.

File: base/reduce.jl (hunk 1 above)
Comment (nalimilan) [reply base/reduce.jl]: Fixed.

File: doc/src/manual/documentation.md (hunk 2 above)
Comment (nalimilan) [reply doc/src/manual/documentation.md]: Fixed.

**Review by andyferris — APPROVED** [review]
Beautiful. Nice work @nalimilan.

**Pull request merged**